src/mimeparser.rs

```diff
@@ -749,33 +744,46 @@ impl<'a> MimeParser<'a> {
                 }));
             }
         }
+        warn!(
+            self.context,
+            "ignoring unknown disposition-notification, Message-Id: {:?}",
+            report_fields.get_first_value("Message-ID").ok()
+        );
 
         Ok(None)
     }
 
-    // Handle reports (mainly MDNs)
+    // Handle reports (only MDNs for now)
     pub fn handle_reports(
         &self,
         from_id: u32,
         sent_timestamp: i64,
-        rr_event_to_send: &mut Vec<(u32, MsgId)>,
         server_folder: impl AsRef<str>,
         server_uid: u32,
     ) {
+        if self.reports.is_empty() {
+            return;
+        }
+        // If a user disabled MDNs we do not show pending incoming ones anymore
+        // but we do want them to potentially get moved from the INBOX still.
+        let mdns_enabled = self.context.get_config_bool(Config::MdnsEnabled);
+
         for report in &self.reports {
-            let mut mdn_consumed = false;
-
-            if let Some((chat_id, msg_id)) = message::mdn_from_ext(
-                self.context,
-                from_id,
-                &report.original_message_id,
-                sent_timestamp,
-            ) {
-                rr_event_to_send.push((chat_id, msg_id));
-                mdn_consumed = true;
+            let mut mdn_recognized = false;
+
+            if mdns_enabled {
```

Comment:
So you decided not to show two ticks if MDNs are disabled?

Reply:
yes, i went with @r10s' view that when you disable read-receipts and then start seeing double-ticks on your messages that you might be wondering  if the setting works. This was also how the old code behaved.